### PR TITLE
Added Brush, Zoom, Context and minor style changes

### PR DESCRIPTION
**Pull request by craigericlewis**
Changes live on [storybook](https://googleinterns.github.io/react-d3-chart/?path=/story/linechart--default)

### CODE REVIEW

**Comment by zhm12359** [comment]
Merge in the other PR first and rebase this one

**Comment by craigericlewis** [comment]
> Merge in the other PR first and rebase this one

Done

File: react-d3-chart/src/components/LineChart/index.tsx [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
 import * as d3 from 'd3';
 import Axis from '../Axis';
 import { LineProps, Dimensions } from '../types';
-import Line from './Line';
-import Bisector from '../Bisector';
+import Line from '../Line';
+import Overlay from '../Overlay';
+import Context from '../Context';
+
+const ContextHeight = 40;
```
Comment (zhm12359) [inline react-d3-chart/src/components/LineChart/index.tsx:9]: CONTEXT_HEIGHT=40

To distinguish constants and Component names.

File: react-d3-chart/src/components/LineChart/index.tsx [hunk 2]
```diff
@@ -1,29 +1,62 @@
...
+interface ScalesState {
+  xScale: d3.ScaleLinear<number, number>;
+}
+
+interface OverlayState {
+  brush: d3.BrushBehavior<unknown>;
+  brushContainer: SVGGElement;
+}
+
+export type State = OverlayState & ScalesState;
```
Comment (zhm12359) [inline react-d3-chart/src/components/LineChart/index.tsx:20]: what does & do in this case?

**Pull request merged**